src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -317,23 +329,27 @@ protected void retrieve(SCMSourceCriteria criteria, @NonNull SCMHeadObserver obs
                                         "!" + m.getIid()
                                 )
                         );
+                        Map<Boolean, Set<ChangeRequestCheckoutStrategy>> strategies = request.getMRStrategies();
+                        boolean fork = !m.getSourceProjectId().equals(m.getTargetProjectId());
                         String originOwner = m.getAuthor().getUsername();
-                        // Origin project name will always the same as the source project name
                         String originProjectPath = projectPath;
-                        Map<Boolean, Set<ChangeRequestCheckoutStrategy>> strategies = request.getMRStrategies();
-                        boolean fork = !gitlabProject.getOwner().getUsername().equals(originOwner);
-                        LOGGER.info(originOwner + " -> " + (request.isMember(originOwner) ? "TRUE" : "FALSE"));
+                        if (fork && !mrSources.containsKey(mr.getSourceProjectId())) {
+                            // This is a hack to get the path with namespace of source project for forked mrs
+                            originProjectPath = gitLabApi.getProjectApi().getProject(m.getSourceProjectId()).getPathWithNamespace();
+                        } else if(fork) {
+                            originProjectPath = mrSources.get(mr.getSourceProjectId());
+                        }
+                        mrSources.put(mr.getSourceProjectId(), originProjectPath);
```

Comment:
This should properly be moved into the if statement where the API call is made?

Reply:
Fixed in 146b535